prototype2/common/Producer.cpp

```diff
@@ -46,68 +35,60 @@ void Producer::event_cb(RdKafka::Event &event) {
 
-  conf = RdKafka::Conf::create(RdKafka::Conf::CONF_GLOBAL);
-  tconf = RdKafka::Conf::create(RdKafka::Conf::CONF_TOPIC);
+  Config.reset(RdKafka::Conf::create(RdKafka::Conf::CONF_GLOBAL));
+  TopicConfig.reset(RdKafka::Conf::create(RdKafka::Conf::CONF_TOPIC));
 
-  if (conf == nullptr) {
+  if (Config == nullptr) {
     LOG(KAFKA, Sev::Error, "Unable to create CONF_GLOBAL object");
     return;
   }
 
-  if (tconf == nullptr) {
+  if (TopicConfig == nullptr) {
     LOG(KAFKA, Sev::Error, "Unable to create CONF_TOPIC object");
     return;
   }
 
-  setConfig("metadata.broker.list", broker);
+  setConfig("metadata.broker.list", Broker);
   setConfig("message.max.bytes", "10000000");
   setConfig("fetch.message.max.bytes", "10000000");
   setConfig("message.copy.max.bytes", "10000000");
   setConfig("queue.buffering.max.ms", "100");
   setConfig("api.version.request", "true");
 
-  if (conf->set("event_cb", this, kafkaErrstr) != RdKafka::Conf::CONF_OK) {
+  if (Config->set("event_cb", this, ErrorMessage) != RdKafka::Conf::CONF_OK) {
     LOG(KAFKA, Sev::Error, "Kafka: unable to set event_cb");
   }
 
-  // if (conf->set("dr_cb", this, kafkaErrstr) != RdKafka::Conf::CONF_OK) {
+  // if (conf->set("dr_cb", this, ErrorMessage) != RdKafka::Conf::CONF_OK) {
   //   LOG(KAFKA, Sev::Error, "Kafka: unable to set dr_cb");
   // }
 
-  producer = RdKafka::Producer::create(conf, kafkaErrstr);
-  if (!producer) {
-    LOG(KAFKA, Sev::Error, "Failed to create producer: {}", kafkaErrstr);
+  KafkaProducer.reset(RdKafka::Producer::create(Config.get(), ErrorMessage));
+  if (!KafkaProducer) {
+    LOG(KAFKA, Sev::Error, "Failed to create producer: {}", ErrorMessage);
     return;
   }
 
-  topic = RdKafka::Topic::create(producer, topicstr, tconf, kafkaErrstr);
-  if (!topic) {
-    LOG(KAFKA, Sev::Error, "Failed to create topic: {}", kafkaErrstr);
+  KafkaTopic.reset(RdKafka::Topic::create(KafkaProducer.get(), TopicName, TopicConfig.get(), ErrorMessage));
+  if (!KafkaTopic) {
+    LOG(KAFKA, Sev::Error, "Failed to create topic: {}", ErrorMessage);
     return;
   }
 }
 
-///
-Producer::~Producer() {
-  delete topic;
-  delete producer;
-  delete tconf;
-  delete conf;
-}
-
 /** called to actually send data to Kafka cluster */
-int Producer::produce(void *buffer, size_t bytes) {
-  if (producer == nullptr || topic == nullptr) {
+int Producer::produce(nonstd::span<const std::uint8_t> Buffer, std::int64_t MessageTimestampMS) {
+  if (KafkaProducer == nullptr || KafkaTopic == nullptr) {
     return RdKafka::ERR_UNKNOWN;
   }
-  int64_t timestamp = time(NULL)*1000;
-  RdKafka::ErrorCode resp = producer->produce(topicString, -1, RdKafka::Producer::RK_MSG_COPY ,
-             buffer, bytes, NULL, 0, timestamp, NULL);
+  
+  RdKafka::ErrorCode resp = KafkaProducer->produce(TopicName, -1, RdKafka::Producer::RK_MSG_COPY ,
+                                              const_cast<std::uint8_t*>(Buffer.data()), Buffer.size_bytes(), NULL, 0, MessageTimestampMS, NULL);
```

Comment:
This line is kind of long. I don't think 80 chars is reasonable in this day and age, but even with my crazy resolution, 145 seems a bit long. I don't usually care about formatting trivialities, but this function could really look a bit better ;)

Reply:
Sure, agreed.